### PR TITLE
Push antithesis images

### PR DESCRIPTION
**Pull request by StephenButtolph**
Introduces new Dockerfiles and a new GitHub Actions workflow for publishing them to the antithesis repository.

Dockerfile Changes:

* [`tests/antithesis/Dockerfile.node`](diffhunk://#diff-5050918789a5172d28bbb20f12f10e670e87c4ce3e5782f5fe39f23825fdfdcaR1-R64): Added a new Dockerfile for building the `avalanche-node` image with Antithesis instrumentation.
* [`tests/antithesis/Dockerfile.workload`](diffhunk://#diff-8250d078286df248f853d40ea54a32ba9bbe8dcaef09dfcb83540e7e9ee9e78fR1-R29): Added a new Dockerfile for building the `workload` image.
* [`tests/antithesis/Dockerfile.config`](diffhunk://#diff-6097f76c273f300260238d12b05c67c440442c8cd9410a0e6f618c4b387e5d8cR1-R3): Added a new Dockerfile for building the `config` image.

Script Changes:

* [`scripts/build_antithesis_workload.sh`](diffhunk://#diff-c192d7950d92de749a316bf9f3b13009200e9f3f1485384431eaf4213ec0a6beR1-R17): Added a new script for building the Antithesis workload.

GitHub Actions Workflow:

* [`.github/workflows/publish_antithesis_images.yml`](diffhunk://#diff-390b2c3f0e3d75fb7aac49d22bb862d06e17a4b3b02c1189f737d2868594bd55R1-R54): Added a new GitHub Actions workflow to build and publish Antithesis images. This workflow is triggered on push to the master branch (or manual dispatch). It builds the three Docker images (`avalanche-node`, `workload`, and `config`) and pushes them to the antithesis Docker registry.

### CODE REVIEW

File: tests/antithesis/Dockerfile.node [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+# ============= Cleanup Stage ================
+FROM debian:11-slim AS execution
+
+# Copy identifying information into the container
+COPY --from=builder /build/commit_hash.txt ./commit_hash.txt
+
+# Copy the antithesis dependencies into the container
+RUN mkdir -p /symbols
+COPY --from=builder /avalanchego_instrumented/symbols /symbols
```
Comment (StephenButtolph) [inline tests/antithesis/Dockerfile.node:58]: Is this the right location for the `symbols`?

File: tests/antithesis/Dockerfile.config
Comment (StephenButtolph) [inline tests/antithesis/Dockerfile.config:1]: I'm not sure how to test this file locally.

**Review by StephenButtolph — COMMENTED** [review]
```bash
$ docker build -t avalanche-node:latest -f ./tests/antithesis/Dockerfile.node .
$ docker build -t workload:latest -f ./tests/antithesis/Dockerfile.workload .
$ cd ./tests/antithesis
$ docker compose up
```
Seems to work as expected ✅

File: .github/workflows/publish_antithesis_images.yml
Comment (StephenButtolph) [inline .github/workflows/publish_antithesis_images.yml:1]: Tested this action (except for the final `Publish images` step) here: https://github.com/ava-labs/avalanchego/actions/runs/8395631342/job/22995314404

**Review by JacobEverly — APPROVED** [review]

**Review by aaronbuchwald — APPROVED** [review]

**Pull request merged**